<launch>

    <!-- <arg name="enable_debug" /> -->
    <arg name="vehicle_name" default="X1"/>
    <arg name="enable_ground_truth" default="false"/>

    <node pkg="nearness_control" type="trajectory_follower" name="trajectory_follower_node">
        <!-- TOPICS -->
        <remap from="trajectory_follower/odometry_map" to="odometry_map"/>
        <remap from="trajectory_follower/cartographer_trajectory" to="pose_graph"/>
        <remap from="trajectory_follower/liosam_trajectory" to="lio_sam/mapping/path"/>
        <remap from="trajectory_follower/follow_traj" to="follow_traj"/>
        <remap from="trajectory_follower/ground_truth_trajectory" to="trajectory_generator/ground_truth_trajectory"/>

        <param name="vehicle_name" value="$(arg vehicle_name)"/>
        <param name="enable_ground_truth" value="$(arg enable_ground_truth)"/>

    </node>

    <node pkg="nearness_control" type="trajectory_generator" name="trajectory_generator_node" if="$(arg enable_ground_truth)">
        <remap from="trajectory_generator/odometry_map" to="odometry_map"/>
    </node>

    <include file="$(find lcd_pkg)/launch/lcd.launch"/>

</launch>
